app.rb

```diff
@@ -3,103 +3,113 @@
 require_relative 'teacher'
 require_relative 'rental'
 
-def list_all_books(books)
-  books.each { |book| puts "Title '#{book.title}', Author #{book.author}" }
-end
+class App
+  attr_accessor :books, :people, :rentals
 
-def list_all_people(people)
-  people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
-end
+  def initialize()
+    @books = []
+    @people = []
+    @rentals = []
+  end
 
-def create_person(people)
-  print 'Do you want to create a student (1) or a teacher (2): '
-  choice = gets.chomp.to_i
-  print 'Age: '
-  age = gets.chomp.to_i
-  print 'Name: '
-  name = gets.chomp
-  case choice
-  when 1
-    print 'Has parent permission? [Y/N]: '
-    permission_input = gets.chomp
-    permission = true if permission_input == 'Y'
-    permission = false if permission_input == 'N'
-    people.push(Student.new(age, name: name, parent_permission: permission))
-  when 2
-    print 'Specialization: '
-    specialization = gets.chomp
-    people.push(Teacher.new(specialization, age, name))
+  def list_all_books()
+    books.each_with_index { |book, index| puts "#{index}) Title: '#{book.title}', Author: #{book.author}" }
   end
-  puts 'Person created successfully.'
-end
 
-def create_book(books)
-  print 'Title: '
-  title = gets.chomp
-  print 'Author: '
-  author = gets.chomp
```

Comment:
- [ ] All requests for inputs should have been placed in the main.rb file and/or classes/methods whose responsibilities are to request input for reasons outlined above.